### DISCUSSION TITLE
Bitmap fonts

### DISCUSSION THREAD

**Discussion by sin-ack** [ghostty-org/ghostty#3543] [Feature Requests, Ideas] [answered]
I've seen various opinions in the Ghostty Discord and the issue tracker about this but no definite answer, so I'd like to ask for one: Will Ghostty ever support bitmap fonts?

Why they should be supported:
- Bitmap fonts look very crisp when used at the correct sizes
- No hinting artifacts
- Lets more people use their favorite fonts when switching from other terminal emulators

Why they should not be supported:
- Additional code complexity

My personal stake in this is that I really enjoy [Terminus TTF](https://files.ax86.net/terminus-ttf/) and it is currently the only thing preventing from completely switching to Ghostty. If Ghostty is willing to support bitmap fonts I would be willing to make the patch happen.

**Comment by rhodes-b** [comment]
#3550 take a look in here on a bit of explanation on why bitmap fonts are not supported

**Comment by ilvez** [comment]
Sad news indeed. Also long time Terminus user here. It's kind of blocker for me as well, but I'm willing to play around, to get benefit of missing features (like icons), is enough to drop the requirement of crisp and clearly readable fonts. Terminus is just so good.

**Comment by sin-ack** [comment]
I had a discussion with Mitchell on the Zig Discord last night and he said he would be willing to accept a patch to make bitmap fonts work out-of-the box "depend[ing] on what depths of changes are required". I will be working on a patch for this.

**Comment by sin-ack** [comment]
I have opened #3837 which allows crisp rendering of bitmap fonts. @ilvez Could you please test and let me know if it fixes it for you as well?

**Reply by ilvez** [reply]
Hey, if you want me to leave feedback under PR then let me know.

This indeed made ghostty show Terminus. I compared it with other terminal, and the rendering is the same. 

Probably you are aware of it, but changing font size is currently not possible. I'm not talking about +/- when ghostty was already running, but even `font-size` option on startup didn't seem to work.

Thank you for the effort and GL on getting it finally in. Terminus support is probably blocker for lots of people.

**Reply by sin-ack** [reply]
Are you using the Terminus TTF variant, or the original Terminus? The TTF variant is a scalable font and thus will render on every size; the non-TTF variant will only render on specific sizes with bitmaps available.

**Reply by ilvez** [reply]
Thanks for the hint. Testing it out with TTF variant.

**Comment by bartman** [comment]
@ilvez did you end up getting Terminus TTF to render "correctly" in ghostty?
I've opened another discussion regarding that here: #4099

**Reply by ilvez** [reply]
I'm currently been using default font, kind of like it. Work hit me, maybe home time to play around in few days.

**Reply by bartman** [reply]
Just to confirm by default font, you mean it's a proportional scaled font, and you get subpixel rendering, right?

I don't think that will work for me. :cry:

**Reply by bartman** [reply]
actually, PR #3837 fixed my issue... https://github.com/ghostty-org/ghostty/discussions/4099#discussioncomment-11729327

![image](https://github.com/user-attachments/assets/4dd85d2f-5ea0-4611-acfc-a90bc57096fe)

**Reply by ilvez** [reply]
I meant conforming to JetBrains Mono. Kind of works well with high resolution displays + zooming is now more natural when jumping between lower resolution and higher resolution screen.

**Comment by sin-ack** [comment, marked as answer]
Closing as #3837 has been merged.
